src/unix/fcitx5/BUILD.bazel

```diff
@@ -6,32 +6,64 @@ load(
 
 package(default_visibility = ["//:__subpackages__"])
 
-mozc_cc_library(
-    name = "mozc_connection",
-    srcs = ["mozc_connection.cc"],
-    hdrs = ["mozc_connection.h"],
-    deps = [
-        ":fcitx_key_util",
-        ":surrounding_text_util",
-        "//base:port",
-        "//base:util",
-        "//base:vlog",
-        "//client",
-        "//protocol:commands_cc_proto",
-    ],
+config_setting(
+    name = "with_server",
+    values = {"define": "server=1"},
+)
+
+config_setting(
+    name = "without_server",
+    values = {"define": "server=0"},
 )
 
 mozc_cc_library(
     name = "mozc_client_pool",
-    srcs = ["mozc_client_pool.cc"],
-    hdrs = ["mozc_client_pool.h"],
+    srcs = [
+        "mozc_client_pool.cc",
+        "mozc_client_interface.cc",
+    ],
+    hdrs = [
+        "mozc_client_pool.h",
+        "mozc_client_interface.h",
+    ],
     deps = [
-        ":mozc_connection",
         "//client:client_interface",
+        "//protocol:commands_cc_proto",
+        "//protocol:config_cc_proto",
         "@fcitx5",
     ],
 )
 
+mozc_cc_library(
+    name = "mozc_client_impl",
```

Comment:
1. create mozc_client_interface target, things like mozc_engine can depend on it.
2. Create two mozc_client_impl target one with server, one without.
3. use select() on target, not on source files or deps, if possible.